api/src/test/java/org/openmrs/module/attachments/obs/ImageAttachmentHandlerTest.java

```diff
@@ -38,50 +43,46 @@ public void tearDown() throws IOException {
 	@Test
 	public void saveComplexData_shouldSaveThumbnailToDisk() throws IOException {
 
-		// Replay
+		// Setup
 		Obs obs = testHelper.saveNormalSizeImageAttachment();
 
-		// Verif
-		MockMultipartFile mpFile = testHelper.getLastSavedTestImageFile();
-		File file = new File(
-				testHelper.getComplexObsDir() + "/" + FilenameUtils.removeExtension(mpFile.getOriginalFilename()) + "_"
-						+ obs.getUuid() + "." + FilenameUtils.getExtension(mpFile.getOriginalFilename()));
-		Assert.assertTrue(file.exists());
-		File thumbnail = new File(testHelper.getComplexObsDir() + "/"
-				+ ImageAttachmentHandler
-						.buildThumbnailFileName(FilenameUtils.removeExtension(mpFile.getOriginalFilename()) + "_"
-								+ obs.getUuid() + "." + FilenameUtils.getExtension(mpFile.getOriginalFilename())));
+		// Verify
+		String originalFilePath = testHelper.getFilePathFromObs(obs);
+		String thumbnailFilePath = appendThumbnailSuffix(originalFilePath);
+
+		File originalFile = new File(testHelper.encode(originalFilePath));;
```

Comment:
Any reason for the double semicolons? :)

Reply:
When you _really_ need to be sure the parser knows that this line has ended 😁